server/resp/src/main/java/org/infinispan/server/resp/RespServer.java

```diff
@@ -168,10 +199,19 @@ public void stop() {
       }
    }
 
+   public CompletionStage<AdvancedCache<byte[], byte[]>> getInitializedCache() {
+      if (!initialized) {
+         return initializeRespCache();
+      }
+      return CompletableFuture.completedFuture(getCache());
+   }
+
    /**
     * Returns the cache being used by the Resp server
     */
    public AdvancedCache<byte[], byte[]> getCache() {
+      if (!initialized)
```

Comment:
Can we not use `EmbeddedCacheManager#isRunning(String)` method instead of the `initialized` variable?

Reply:
Let me check, I think it should be possible.

Reply:
I've updated this. The RESP initialization had a little bit more work. The method was initializing both the cache AND the server variables. I've moved the initialization around, so the server state is initialized after the cache manager starts, and the cache is initialized only if not already created (cluster-wide).